packages/redux/src/raffles/reducer.ts

```diff
@@ -0,0 +1,180 @@
+    case actionTypes.FETCH_RAFFLE_FAILURE:
+    case actionTypes.FETCH_RAFFLE_ESTIMATION_SUCCESS:
+    case actionTypes.FETCH_RAFFLE_ESTIMATION_FAILURE:
+    case actionTypes.FETCH_RAFFLE_PARTICIPATION_SUCCESS:
+    case actionTypes.FETCH_RAFFLE_PARTICIPATION_FAILURE:
+    case actionTypes.CREATE_RAFFLE_PARTICIPATION_SUCCESS:
+    case actionTypes.CREATE_RAFFLE_PARTICIPATION_FAILURE:
+    case actionTypes.MODIFY_RAFFLE_PARTICIPATION_SUCCESS:
+    case actionTypes.MODIFY_RAFFLE_PARTICIPATION_FAILURE:
+      return INITIAL_STATE.isLoading;
+    default:
+      return state;
+  }
+};
+
+const result = (state = INITIAL_STATE.result, action: AnyAction) => {
+  switch (action.type) {
+    case actionTypes.FETCH_RAFFLES_SUCCESS:
+      return action.payload.result;
+    default:
+      return state;
+  }
+};
+
+const raffle = (state = INITIAL_STATE.raffle, action: AnyAction) => {
+  if (action.type === actionTypes.FETCH_RAFFLE_SUCCESS) {
+    return action.payload.result;
+  }
+  return state;
+};
+
+const resetEntitiesStateReducer = (
+  state: NonNullable<StoreState['entities']>,
+) => {
+  const { raffles, ...rest } = state;
+
+  return rest;
+};
+
+export const entitiesMapper = {
+  [actionTypes.FETCH_RAFFLE_ESTIMATION_SUCCESS]: (
+    state: NonNullable<StoreState['entities']>,
+    action: AnyAction,
+  ) => {
+    const { raffleId, estimation } = action.payload;
+    if (!estimation) {
+      return state;
+    }
+    return {
+      ...state,
+      raffles: {
+        ...state.raffles,
+        [raffleId]: {
+          ...state?.raffles?.[raffleId],
+          estimation,
+        },
+      },
+    };
+  },
+  [actionTypes.FETCH_RAFFLE_PARTICIPATION_SUCCESS]: (
+    state: NonNullable<StoreState['entities']>,
+    action: AnyAction,
+  ) => {
+    const { raffleId, participation } = action.payload;
+    if (!participation) {
+      return state;
+    }
+    return {
+      ...state,
+      raffles: {
+        ...state.raffles,
+        [raffleId]: {
+          ...state?.raffles?.[raffleId],
+          participations: participation.result,
+        },
+      },
+      ...participation.entities,
+    };
+  },
+  [actionTypes.CREATE_RAFFLE_PARTICIPATION_SUCCESS]: (
+    state: NonNullable<StoreState['entities']>,
+    action: AnyAction,
+  ) => {
+    const { raffleId, participation } = action.payload;
+    if (!participation) {
+      return state;
+    }
+    return {
+      ...state,
+      raffles: {
+        ...state.raffles,
+        [raffleId]: {
+          ...state?.raffles?.[raffleId],
+          participations: [
+            ...(state?.raffles?.[raffleId]?.participations as string[]),
+            participation.id,
+          ],
+        },
+      },
+      raffleParticipations: {
+        ...state?.raffleParticipations,
+        [participation.id]: participation,
+      },
+    };
+  },
+  [actionTypes.RESET_RAFFLES_STATE]: resetEntitiesStateReducer,
+};
+
+export const getError = (state: T.RafflesState): T.RafflesState['error'] =>
```

Comment:
No need to declare the return type since TS can infer it.

```suggestion
export const getError = (state: T.RafflesState) =>
```

Reply:
done.